src/AppBundle/Repository/EventRepository.php

```diff
@@ -396,6 +396,135 @@ private function getUsernamesSql(Event $event): string
         return ltrim(implode(',', $usernames), ',');
     }
 
+    /**
+     * @param Event $event
+     * @return int[]
+     */
+    public function getBytesChanged(Event $event): array
+    {
+        $firstLast = $this->getFirstLastRevisions($event);
+
+        $result = [];
+        foreach ($firstLast as $name => $pages) {
+            $wiki = $event->getWikiByDomain($name);
+
+            /** @var EventWikiRepository $eventWikiRepo */
+            $eventWikiRepo = $this->em->getRepository('Model:EventWiki');
+            $eventWikiRepo->setContainer($this->container);
+
+            $dbName = $eventWikiRepo->getDbNameFromDomain($wiki->getDomain());
+
+            $first = [];
+            $last = [];
+            foreach ($pages as $info) {
+                $first[] = (int)$info['first']['id'];
+                $last[] = (int)$info['last']['id'];
+            }
+
+            $result[$name] = $this->getBytesAfterEvent($dbName, $last)
+                - $this->getBytesBeforeEvent($dbName, $first);
+        }
+
+        return $result;
+    }
+
+    /**
+     * Returns an SQL-friendly stringified list of identifiers
+     *
+     * @param int[] $ids
+     * @return string
+     */
+    private function idList(array $ids): string
+    {
+        $connection = $this->getReplicaConnection();
+
+        return implode(', ', array_map(function ($id) use ($connection) {
+            return $connection->quote($id, \PDO::PARAM_INT);
+        }, $ids));
+    }
+
+    /**
+     * @param string $dbName
+     * @param int[] $revIds
+     * @return int
+     */
+    private function getBytesAfterEvent(string $dbName, array $revIds): int
+    {
+        $connection = $this->getReplicaConnection();
+        $revisionTable = $this->getTableName('revision');
+
+        $sql = "SELECT /* BytesAfterEvent */ SUM(IFNULL(rev_len, 0)) AS result
+            FROM $dbName.$revisionTable
+            WHERE rev_id IN ({$this->idList($revIds)})";
+
+        return (int)$connection->executeQuery($sql)->fetchColumn();
+    }
+
+    /**
+     * @param string $dbName
+     * @param int[] $revIds
+     * @return int
+     */
+    private function getBytesBeforeEvent(string $dbName, array $revIds): int
+    {
+        $connection = $this->getReplicaConnection();
+        $revisionTable = $this->getTableName('revision');
+
+        $sql = "SELECT /* BytesBeforeEvent */ SUM(IFNULL(t2.rev_len, 0)) AS result
+            FROM $dbName.$revisionTable t1
+                LEFT JOIN $dbName.$revisionTable t2
+                    ON t2.rev_id = t1.rev_parent_id
+            WHERE t1.rev_id IN ({$this->idList($revIds)})";
+
+        return (int)$connection->executeQuery($sql)->fetchColumn();
+    }
+
+    /**
+     * @param Event $event
+     * @return mixed[]
+     */
+    private function getFirstLastRevisions(Event $event): array
+    {
+        /** @var string[][] $revs */
+        $revs = $this->getRevisions($event);
```

Comment:
I imagine going through every edit is somewhat slow... I wonder if there's a way to get the length of the first/last revision using dedicated SQL queries? We have the page IDs on hand, so you could loop through those and make queries accordingly.

Reply:
I was assuming this was going to be the intended use case for revision IDs that we store.

Reply:
I'll reply here too... I don't think the plan is to store revision IDs, if that's what you mean? Only page IDs. So I envisioned using one or two queries to fetch `rev_len` from the first/last revision with `rev_page IN (...)`.